examples/fly_qgc_mission/fly_qgc_mission.cpp

```diff
@@ -130,44 +124,26 @@ int main(int argc, char** argv)
     std::cout << "Found " << import_res.second.mission_items.size()
               << " mission items in the given QGC plan." << std::endl;
 
-    {
-        std::cout << "Uploading mission..." << std::endl;
-        // Wrap the asynchronous upload_mission function using std::future.
-        auto prom = std::make_shared<std::promise<Mission::Result>>();
-        auto future_result = prom->get_future();
-        mission.upload_mission_async(
-            import_res.second, [prom](Mission::Result result) { prom->set_value(result); });
-
-        const Mission::Result result = future_result.get();
-        handle_mission_err_exit(result, "Mission upload failed: ");
-        std::cout << "Mission uploaded." << std::endl;
-    }
+    std::cout << "Uploading mission..." << std::endl;
+    MissionRaw::Result result = mission_raw.upload_mission(import_res.second.mission_items);
+    handle_mission_err_exit(result, "Mission upload failed: ");
+    std::cout << "Mission uploaded." << std::endl;
 
     std::cout << "Arming..." << std::endl;
     const Action::Result arm_result = action.arm();
     handle_action_err_exit(arm_result, "Arm failed: ");
     std::cout << "Armed." << std::endl;
 
     // Before starting the mission subscribe to the mission progress.
-    mission.subscribe_mission_progress([](Mission::MissionProgress mission_progress) {
-        std::cout << "Mission status update: " << mission_progress.current << " / "
+    mission_raw.subscribe_mission_progress([](MissionRaw::MissionProgress mission_progress) {
+        std::cout << "Mission progress update: " << mission_progress.current << " / "
                   << mission_progress.total << std::endl;
     });
 
-    {
-        std::cout << "Starting mission." << std::endl;
-        auto prom = std::make_shared<std::promise<Mission::Result>>();
-        auto future_result = prom->get_future();
-        mission.start_mission_async([prom](Mission::Result result) {
-            prom->set_value(result);
-            std::cout << "Started mission." << std::endl;
-        });
-
-        const Mission::Result result = future_result.get();
-        handle_mission_err_exit(result, "Mission start failed: ");
-    }
+    result = mission_raw.start_mission();
+    handle_mission_err_exit(result, "Mission start failed: ");
 
-    while (!mission.is_mission_finished().second) {
+    while (telemetry.armed()) {
         sleep_for(seconds(1));
     }
```

Comment:
I guess I'd rather use `telemetry.subscribe_armed()` and a fut/prom, but that's fine like this, too

Reply:
Hehe, it's simple enough. And I don't care too much for this example. There is lots that could be improved in my opinion.